src/v/kafka/protocol/CMakeLists.txt

```diff
@@ -1,10 +1,38 @@
 add_subdirectory(schemata)
 
+set(message_gen ${CMAKE_CURRENT_SOURCE_DIR}/schemata/generator.py)
+set(message_srcs)
+
+foreach(schema ${schemata})
+  get_filename_component(msg_name ${schema} NAME_WE)
+  set(schema_src ${CMAKE_CURRENT_SOURCE_DIR}/schemata/${schema})
+  set(msg_srcs
+    "${CMAKE_CURRENT_BINARY_DIR}/${msg_name}.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/${msg_name}.cc")
+  list(APPEND messages_srcs ${msg_srcs})
```

Comment:
the variable you are appending to is `messages_srcs` but the variable defined above and used below in the library is `message_srcs` (singular)

Reply:
if you tried to build without a make clean it may still have succeeded.